addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/VoiceRSSCloudImplementation.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.voice.voicerss.internal.cloudapi;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements the Cloud service from VoiceRSS. For more information,
+ * see API documentation at http://www.voicerss.org/api/documentation.aspx.
+ * 
+ * Current state of implementation:
+ * <ul>
+ * <li>Only EN and DE languages supported</li>
+ * <li>Only default voice supported with good audio quality</li>
+ * <li>Only MP3 audio format supported</li>
+ * <li>It uses HTTP and not HTTPS (for performance reasons)</li>
+ * </ul>
+ * 
+ * @author Jochen Hiller - Initial contribution
+ */
+public class VoiceRSSCloudImplementation implements VoiceRSSCloudAPI {
+
+	private final Logger logger = LoggerFactory.getLogger(VoiceRSSCloudImplementation.class);
+
+	/**
+	 * Will support only "MP3" for the moment.
+	 */
+	@Override
+	public Set<String> getAvailableAudioFormats() {
+		Set<String> formats = new HashSet<String>();
+		formats.add("MP3");
+		return formats;
+	}
+
+	/**
+	 * Will support only 3 locales for the moment.
+	 */
+	@Override
+	public Set<Locale> getAvailableLocales() {
+		Set<Locale> locales = new HashSet<Locale>();
```

Comment:
You should instantiate this only once as a field in the class and not on every call of get.